addons/account_peppol/wizard/account_move_send.py

```diff
@@ -0,0 +1,177 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from base64 import b64encode
+
+from odoo import api, fields, models, _
+from odoo.addons.account_edi_proxy_client.models.account_edi_proxy_user import AccountEdiProxyError
+from odoo.exceptions import UserError
+
+
+class AccountMoveSend(models.Model):
+    _inherit = 'account.move.send'
+
+    checkbox_send_peppol = fields.Boolean(
+        string='Send via PEPPOL',
+        help='Send the invoice via PEPPOL',
+    )
+    peppol_proxy_state = fields.Selection(related='company_id.account_peppol_proxy_state')
+    peppol_warning = fields.Char(
+        string="Warning",
+        compute="_compute_peppol_warning",
+    ) # technical field needed for computing a warning text about the peppol configuration
+
+    # -------------------------------------------------------------------------
+    # COMPUTE METHODS
+    # -------------------------------------------------------------------------
+
+    @api.depends('checkbox_send_peppol')
+    def _compute_checkbox_ubl_cii_xml(self):
+        # Extends account.move.send wizard - auto check checkbox_ubl_cii_xml
+        # if checkbox_send_peppol is selected
+        super()._compute_checkbox_ubl_cii_xml()
+        for wizard in self:
+            if wizard.checkbox_send_peppol and not wizard.checkbox_ubl_cii_xml:
+                wizard.checkbox_ubl_cii_xml = True
+
+    @api.depends('checkbox_send_peppol')
+    def _compute_mail_attachments_widget(self):
+        # EXTENDS 'account' - add depends
+        super()._compute_mail_attachments_widget()
+
+    @api.depends('move_ids')
+    def _compute_peppol_warning(self):
+        for wizard in self:
+            invalid_partners = wizard.move_ids.partner_id.filtered(
+                lambda partner: not partner.account_peppol_is_endpoint_valid or partner.ubl_cii_format not in ('ubl_bis3', 'xrechnung', 'nlcius', 'ubl_a_nz', 'ubl_sg'))
+            if not invalid_partners:
+                self.peppol_warning = False
+            else:
+                names = ', '.join([partner.name for partner in invalid_partners[:5]])
+                self.peppol_warning = _("The following partners are not correctly configured to receive Peppol documents. "
+                                        "Please check and verify their Peppol endpoint and the Electronic Invoicing format: "
+                                        "%s", names)
+
+    # -------------------------------------------------------------------------
+    # ATTACHMENTS
+    # -------------------------------------------------------------------------
+
+    @api.model
+    def _get_linked_attachments(self, move):
+        # EXTENDS 'account'
+        if self.checkbox_send_peppol:
+            return super()._get_linked_attachments(move) - move.ubl_cii_xml_id
+        return super()._get_linked_attachments(move)
```

Comment:
You can remove this code. You must not access to a field on a api.model method.

Reply:
But why is it `api.model` in the first place?